lib/events.ml

```diff
@@ -0,0 +1,381 @@
+open Common
+open Lwt.Infix
+
+module Smap = Map.Make(String)
+
+module Message = struct
+  type recipients =
+    | Stream of int * string
+    | Private of int list
+
+  type flag =
+    | Read 
+    | Starred
+    | Collapsed
+    | Mentioned
+    | Wildcard_mentioned
+    | Has_alert_word
+    | Historical
+
+  let flag =
+    Json_encoding.(
+      string_enum [
+        "read", Read;
+        "starred", Starred;
+        "collapsed", Collapsed;
+        "mentioned", Mentioned;
+        "wildcard_mentioned", Wildcard_mentioned;
+        "has_alert_word", Has_alert_word;
+        "historical", Historical;
+      ]
+    )
+
+  type t = { 
+    id: int ;
+    sender_id: int ;
+    sender_email: string ;
+    sender_full_name: string;
+    recipients: recipients ;
+    content : string ;
+    flags : flag list ;
+  }
+
+  let content { content; _ } = content
+
+  let sender_mention ?(silent = false) { sender_full_name; sender_id; _ } =
+    Format.asprintf "@%s**%s|%d**" (if silent then "_" else "") sender_full_name sender_id
+
+  let sender_destination { sender_id; _ } = Messages.private_ids [sender_id]
+
+  let destination { recipients; _ } =
+    match recipients with
+    | Stream (stream_id, topic) -> Messages.stream_id stream_id topic
+    | Private ids -> Messages.private_ids ids
+
+  (* Message filters *)
+
+  let is_trusted ?(trusted_ids = []) ?(trusted_emails = []) { sender_id; sender_email; _ } =
+    List.mem sender_id trusted_ids || List.mem sender_email trusted_emails
+
+  let has_flag flag m =
+    if List.mem flag m.flags then true else false
```

Comment:
You can remove the `if` :

```ocaml
 let has_flag flag m =
    List.mem flag m.flags
```

Reply:
Ah good catch, this used to return an option. I will remove the `if`